### PR TITLE
[CFM-1059] Update Voices & Views styling to match new theming

### PR DESCRIPTION
**Pull request by nirgn975**
https://issuetracker.amplexor.com/jira/browse/CFM-1059

### CODE REVIEW

**Comment by nirgn975** [comment]
2004_indian_ocean_earthquake_and_tsunami___capacity4more

**Comment by nirgn975** [comment]
the_exothermic_chemical_process_of_combustion___capacity4more

**Comment by nirgn975** [comment]
@ordavidil Ready for review.

File: project/profiles/capacity4more/modules/c4m/content/c4m_content_article/c4m_content_article.module [hunk 1]
```diff
@@ -146,3 +146,27 @@ function c4m_content_article_c4m_offcanvas_trigger_label_left_alter(&$trigger_la
     $trigger_label['icon'] = 'filter';
   }
 }
+
+/**
+ * Implements hook_preprocess_HOOK().
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/content/c4m_content_article/c4m_content_article.module]: remove the `_HOOK` since this is a regular _hook_preprocess_

File: project/profiles/capacity4more/modules/c4m/content/c4m_content_article/c4m_content_article.module [hunk 2]
```diff
@@ -146,3 +146,27 @@ function c4m_content_article_c4m_offcanvas_trigger_label_left_alter(&$trigger_la
...
+function c4m_content_article_preprocess(&$variables) {
+  if ($variables['theme_hook_original'] != 'page') {
+    return;
+  }
+
+  if ($variables['node']->type != 'article') {
+    return;
+  }
+
+  // Get the tree array, and slice it to start from the second child.
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/content/c4m_content_article/c4m_content_article.module]: Please elaborate ...

File: project/profiles/capacity4more/themes/c4m/kapablo/build/src/stylesheets/components/_groups.scss [hunk 3]
```diff
@@ -461,16 +461,37 @@
...
+    padding: 10px;
+
     h3 {
+      text-transform: uppercase;
       margin-bottom: 5px;
     }
   }
 }
 
+h1 {
```
Comment (ordavidil) [inline project/profiles/capacity4more/themes/c4m/kapablo/build/src/stylesheets/components/_groups.scss]: @nirgn975 
It seems to general ...   can't we be more specific with the rule over here ?

File: project/profiles/capacity4more/modules/c4m/content/c4m_content_article/c4m_content_article.module [hunk 4]
```diff
@@ -146,3 +146,27 @@ function c4m_content_article_c4m_offcanvas_trigger_label_left_alter(&$trigger_la
...
+  $vocab_geo = $variables['node']->c4m_vocab_geo['und'];
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/content/c4m_content_article/c4m_content_article.module]: `'und'` => `LANGUAGE_NONE`

File: project/profiles/capacity4more/themes/c4m/kapablo/build/src/stylesheets/components/_groups.scss (hunk 3 above)
Comment (nirgn975) [inline project/profiles/capacity4more/themes/c4m/kapablo/build/src/stylesheets/components/_groups.scss]: This is `h1.page-header` - there is only one class like this in the page. But I can be a little more specific if you wish.

File: project/profiles/capacity4more/themes/c4m/kapablo/build/src/stylesheets/components/_groups.scss [hunk 5]
```diff
@@ -465,6 +465,7 @@
 
   .social-actions-share {
     margin-top: 20px;
+
```
Comment (ordavidil) [inline project/profiles/capacity4more/themes/c4m/kapablo/build/src/stylesheets/components/_groups.scss]: @nirgn975 
revert this one.

**Comment by nirgn975** [comment]
@ordavidil Ready for review.

**Comment by nirgn975** [comment]
@ordavidil Ready for review.

**Comment by jover** [comment]
Did some QC, but it's hard to know what's right or wrong without a design.

First of all, I think that the uppercase headings (title of the article and the blocks in the sidebar) to be found in the wireframes are not to be implemented as it's not implemented elsewhere on the website so far.

What I do see is that the title of the article should appear underneath the banner image, which is not the case.
So I guess this is already a point for improvement.

Besides that, also a programmatic issue appears.
In c4m_content_article_preprocess(), an error appears when $variables['node'] is not set.

File: project/profiles/capacity4more/modules/c4m/content/c4m_content_article/c4m_content_article.module [hunk 6]
```diff
@@ -146,3 +146,29 @@ function c4m_content_article_c4m_offcanvas_trigger_label_left_alter(&$trigger_la
...
+
+/**
+ * Implements hook_preprocess().
+ */
+function c4m_content_article_preprocess(&$variables) {
+  if ($variables['theme_hook_original'] != 'page') {
+    return;
+  }
+
+  if ($variables['node']->type != 'article') {
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/content/c4m_content_article/c4m_content_article.module]: @nirgn975 
```php
if (empty($variables['node'])) {
  return;
}
```

File: project/profiles/capacity4more/themes/c4m/kapablo/build/src/stylesheets/components/_vocies_and_views.scss [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+    }
+  }
+
+  .group-node-meta {
+    .field-name-c4m-vocab-topic,
+    .field-name-c4m-vocab-geo {
+      padding: 10px;
+      margin-bottom: 2em;
+
+      h3 {
```
Comment (ordavidil) [inline project/profiles/capacity4more/themes/c4m/kapablo/build/src/stylesheets/components/_vocies_and_views.scss]: @nirgn975 
remove this

File: project/profiles/capacity4more/themes/c4m/kapablo/build/src/stylesheets/components/_vocies_and_views.scss [hunk 8]
```diff
@@ -0,0 +1,33 @@
+.node_type-article {
+  h1 {
```
Comment (ordavidil) [inline project/profiles/capacity4more/themes/c4m/kapablo/build/src/stylesheets/components/_vocies_and_views.scss]: @nirgn975 
remove this.

File: project/profiles/capacity4more/themes/c4m/kapablo/build/src/stylesheets/components/_vocies_and_views.scss [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+
+  .field-name-c4m-ds-content-actions {
+    margin: 15px 0 15px;
+
+    .social-actions-share {
+      margin-top: 20px;
+      padding: 10px;
+
+      h3 {
+        text-transform: uppercase;
```
Comment (ordavidil) [inline project/profiles/capacity4more/themes/c4m/kapablo/build/src/stylesheets/components/_vocies_and_views.scss]: @nirgn975 
remove this

File: project/profiles/capacity4more/modules/c4m/content/c4m_content_article/c4m_content_article.module [hunk 10]
```diff
@@ -155,7 +155,7 @@ function c4m_content_article_preprocess(&$variables) {
     return;
   }
 
-  if ($variables['node']->type != 'article') {
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/content/c4m_content_article/c4m_content_article.module]: @nirgn975 
you should've leave this check and just to add another one ...

**Comment by ordavidil** [comment]
@nirgn975 
please re-roll

File: project/profiles/capacity4more/modules/c4m/content/c4m_content_article/c4m_content_article.module (hunk 2 above)
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/content/c4m_content_article/c4m_content_article.module]: @nirgn975  ^^
please describe what's on the first child that is not needed the `<i>` tag, and so on ...

**Comment by nirgn975** [comment]
@ordavidil Ready for review.

**Comment by ordavidil** [comment]
No it's no,
https://github.com/capacity4dev/capacity4more/pull/1027/files#r71666380

**Comment by nirgn975** [comment]
@ordavidil Ready for review.

**Comment by ordavidil** [comment]
@nirgn975 
new screenshots, please

File: project/profiles/capacity4more/modules/c4m/content/c4m_content_article/c4m_content_article.module [hunk 11]
```diff
@@ -146,3 +146,35 @@ function c4m_content_article_c4m_offcanvas_trigger_label_left_alter(&$trigger_la
...
+
+  // Go through the array and add an <i> tag, before the text, to each of them.
+  foreach ($vocab_geo as $geo) {
+    $geo_taxonomy = taxonomy_term_load($geo['tid']);
+
+    // Add an <i> tag to the 'Regions & Countries' tree children.
+    $geo_taxonomy->name = ' <i class="fa fa-caret-right" aria-hidden="true"></i> ' . $geo_taxonomy->name;
+  }
+
+  $variables['hide_title'] = TRUE;
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/content/c4m_content_article/c4m_content_article.module]: Add a comment

File: project/profiles/capacity4more/modules/c4m/content/c4m_content_article/c4m_content_article.module [hunk 12]
```diff
@@ -146,3 +146,35 @@ function c4m_content_article_c4m_offcanvas_trigger_label_left_alter(&$trigger_la
     $trigger_label['icon'] = 'filter';
   }
 }
+
+/**
+ * Implements hook_preprocess().
+ */
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/content/c4m_content_article/c4m_content_article.module]: Add docs

File: project/profiles/capacity4more/themes/c4m/kapablo/templates/page.tpl.php [hunk 13]
```diff
@@ -249,11 +249,13 @@
             ); ?></div>
         <?php endif; ?>
         <a id="main-content"></a>
-        <?php print render($title_prefix); ?>
-        <?php if (!empty($title)): ?>
-          <h1 class="page-header"><?php print $title; ?></h1>
+        <?php if (!isset($hide_title)): ?>
```
Comment (ordavidil) [inline project/profiles/capacity4more/themes/c4m/kapablo/templates/page.tpl.php]:
```php
<?php if (!empty($hide_title)): ?>
```

File: project/profiles/capacity4more/modules/c4m/content/c4m_content_article/c4m_content_article.module [hunk 14]
```diff
@@ -146,3 +146,35 @@ function c4m_content_article_c4m_offcanvas_trigger_label_left_alter(&$trigger_la
...
+  // Since the first element on the tree is actually the title we remove
+  // it because it should not have the arrow icon next to it (the <i> tag).
+  $vocab_geo = $variables['node']->c4m_vocab_geo[LANGUAGE_NONE];
+  $vocab_geo = array_slice($vocab_geo, 1);
+
+  // Go through the array and add an <i> tag, before the text, to each of them.
+  foreach ($vocab_geo as $geo) {
+    $geo_taxonomy = taxonomy_term_load($geo['tid']);
+
+    // Add an <i> tag to the 'Regions & Countries' tree children.
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/content/c4m_content_article/c4m_content_article.module]: Fix the comment to explain about what's actually this `<i>` represents.

File: project/profiles/capacity4more/modules/c4m/content/c4m_content_article/c4m_content_article.module [hunk 15]
```diff
@@ -146,3 +146,35 @@ function c4m_content_article_c4m_offcanvas_trigger_label_left_alter(&$trigger_la
...
+  if ($variables['node']->type != 'article') {
+    return;
+  }
+
+  // Since the first element on the tree is actually the title we remove
+  // it because it should not have the arrow icon next to it (the <i> tag).
+  $vocab_geo = $variables['node']->c4m_vocab_geo[LANGUAGE_NONE];
+  $vocab_geo = array_slice($vocab_geo, 1);
+
+  // Go through the array and add an <i> tag, before the text, to each of them.
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/content/c4m_content_article/c4m_content_article.module]: Fix the comment

File: project/profiles/capacity4more/modules/c4m/content/c4m_content_article/c4m_content_article.module (hunk 14 above)
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/content/c4m_content_article/c4m_content_article.module]: Also, there is a space at the beginning of the string, is that on purpose ?
if no then remove it, if yes then explain why...

File: project/profiles/capacity4more/themes/c4m/kapablo/templates/page.tpl.php (hunk 13 above)
Comment (nirgn975) [inline project/profiles/capacity4more/themes/c4m/kapablo/templates/page.tpl.php]: `empty` remove the title from `node/add/feed`, and `isset` isn't.
I think I should still use `isset` because `empty` return true if the variable is an empty string, false, array(), NULL, “0?, 0, and an unset variable. and `isset` just when the variable is not null - which is (I think) more appropriate here.

File: project/profiles/capacity4more/themes/c4m/kapablo/templates/page.tpl.php [hunk 16]
```diff
@@ -252,8 +252,8 @@
         <?php print render($title_prefix); ?>
         <?php if (!empty($title)): ?>
           <h1 class="page-header"><?php print $title; ?></h1>
-        <?php endif; ?>
         <?php print render($title_suffix); ?>
+        <?php endif; ?>
```
Comment (ordavidil) [inline project/profiles/capacity4more/themes/c4m/kapablo/templates/page.tpl.php]: @nirgn975 
please revert the changes on this template.

File: project/profiles/capacity4more/themes/c4m/kapablo/build/src/stylesheets/kapablo.scss [hunk 17]
```diff
@@ -36,6 +36,7 @@
 @import "components/users";
 @import "components/views";
 @import "components/vocab";
+@import "components/vocies_and_views";
```
Comment (ordavidil) [inline project/profiles/capacity4more/themes/c4m/kapablo/build/src/stylesheets/kapablo.scss]: @nirgn975 
there is a typo over here, should be `voices_and_views`.

File: project/profiles/capacity4more/themes/c4m/kapablo/build/src/stylesheets/components/_vocies_and_views.scss [hunk 18]
```diff
@@ -0,0 +1,23 @@
+.node_type-article {
+
+  .group-node-meta {
+    .field-name-c4m-vocab-topic,
```
Comment (ordavidil) [inline project/profiles/capacity4more/themes/c4m/kapablo/build/src/stylesheets/components/_vocies_and_views.scss]: Add a line spacing above

**Comment by ordavidil** [comment]
@nirgn975 
If you finished, please attach new screenshots, and assign it to me when travis is green

File: project/profiles/capacity4more/modules/c4m/content/c4m_content_article/c4m_content_article.module [hunk 19]
```diff
@@ -180,3 +180,34 @@ function c4m_content_article_field_formatter_view($entity_type, $entity, $field,
   }
   return $element;
 }
+
+/**
+ * Implements hook_node_view_alter().
+ *
+ * Style the 'Regions & Countries' section in the right dashboard.
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/content/c4m_content_article/c4m_content_article.module]: @nirgn975 
I think we should mention "Voices & Views" somewhere in these docs.

**Comment by ordavidil** [comment]
Still needs to fix the tree

**Comment by ordavidil** [comment]
@anvmn 
please add screenshots

**Comment by anvmn** [comment]
selection_063
selection_064

Not displayed, per design:
selection_065
